path/path.go

```diff
@@ -39,13 +48,36 @@ func (s *Resolver) ResolvePath(fpath string) (*merkledag.Node, error) {
 
 	// if nothing, bail.
 	if len(parts) == 0 {
-		return nil, fmt.Errorf("ipfs path must contain at least one component")
+		return nil, nil, fmt.Errorf("ipfs path must contain at least one component")
 	}
 
 	// first element in the path is a b58 hash (for now)
 	h, err := mh.FromB58String(parts[0])
 	if err != nil {
 		log.Debug("given path element is not a base58 string.\n")
+		return nil, nil, err
+	}
+
+	return h, parts[1:], nil
+}
+
+// ResolvePath fetches the node for given path. It returns the last item
+// returned by ResolvePathComponents.
+func (s *Resolver) ResolvePath(fpath string) (*merkledag.Node, error) {
+	nodes, err := s.ResolvePathComponents(fpath)
+	if nodes == nil {
+		return nil, err
+	} else {
+		return nodes[len(nodes)-1], err
+	}
```

Comment:
always check for `err != nil` because the other thing may return `err` even with partially valid output (and in most cases we'll want to suppress valid outputs when there are errors). I get the intent to check for nil on the slice... maybe:
```
if err != nil || nodes ==  nil {
    return nil, err
}
return nodes[len(nodes)-1], nil
```